core/errors.go

```diff
@@ -0,0 +1,206 @@
+package core
+
+// (C) Copyright IBM Corp. 2023.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import (
+  "crypto/sha256"
+  "encoding/hex"
+  "fmt"
+  "github.com/ghodss/yaml"
+  "strings"
+)
+
+// Type definitions
+
+type Problem interface {
+  GetConsoleMessage() string
+  GetDebugMessage() string
+  GetID() string
+}
+
+type IBMError struct {
+  Summary string // required
```

Comment:
We probably should also add comments to each of the public fields in the various types.

Reply:
Agreed